Added Adafruit PyGamer Target

APDevice opened the pull request:
I have added support for the Adafruit PyGamer. The board is nearly identical to the PyBadge, just with a few changes and additions. Chiefly the directional buttons have been replaced with an analog stick and there is an microSD card slot for additional storage.

This is my first attempt to add a device, though I did test it on the hardware to make sure that I had assigned the analog stick correctly


Schematic: https://learn.adafruit.com/assets/76005

APDevice committed:
PyGamer

Added support for the Adafruit Pygamer, a varient of the PyBadge.

APDevice committed:
changed SDCS to SD_CS

Revised the name of variable to be in line with others of the same type.

APDevice committed:
Revert "changed SDCS to SD_CS"

This reverts commit beb1419bfafeb72002c219e94fae35b972553c41.

APDevice committed:
Revert "Revert "changed SDCS to SD_CS""

This reverts commit 7e3be400b00e1edc9108149d976f5f9cda208553.

APDevice committed:
Delete .gitignore

APDevice committed:
Update board_pygamer.go

APDevice committed:
Merge branch 'master' of https://github.com/APDevice/tinygo

deadprogram commented:
Hi @APDevice first of all, thanks for the contribution.

I have changed the branch to be against the `dev` branch as mentioned here: https://github.com/tinygo-org/tinygo/blob/master/CONTRIBUTING.md#how-to-use-our-github-repository

I do not think you meant to delete the `.gitignore` file, did you? We need that file, you probably want to remove that commit and then force push back to your feature branch.

Thank you!

APDevice committed:
restore .github

APDevice commented:
> Hi @APDevice first of all, thanks for the contribution.
> 
> I have changed the branch to be against the `dev` branch as mentioned here: https://github.com/tinygo-org/tinygo/blob/master/CONTRIBUTING.md#how-to-use-our-github-repository
> 
> I do not think you meant to delete the `.gitignore` file, did you? We need that file, you probably want to remove that commit and then force push back to your feature branch.
> 
> Thank you!

AH yes, sorry. I got a little confused with that. I believe I have fixed it now!

deadprogram commented on src/machine/board_pygamer.go:
```diff
@@ -0,0 +1,147 @@
+	SCL_PIN = PA13 // SCL: SERCOM2/PAD[1]
+)
+
+// I2C on the ItsyBitsy M4.
```
That should say "PyGamer" instead of  "ItsyBitsy M4" right?

APDevice replied:
Oh, I hadn’t even touched that; Since this is a variant of the pyBadge, I just copied and modified that. Carrying over all the quirks in the process, it seems. I can go ahead and fix it here though.

deadprogram commented:
Please also add a "smoke test" to the Makefile like this: https://github.com/tinygo-org/tinygo/blob/dev/Makefile#L296-L297

Thank you!

APDevice committed:
Corrected reference to board in comment

APDevice committed:
added smoke test for pygamer

APDevice committed:
Update Makefile

APDevice committed:
Update Makefile

APDevice committed:
Merge branch 'master' of https://github.com/APDevice/tinygo

APDevice committed:
resolved conflict with dev version

APDevice commented:
So I added the smoke test, but now it's saying it's conflicted with the dev makefile, and the error just suggests that the conflict is with a blank line. I’m not sure how to resolve the issue.

deadprogram commented:
If you had originally created your feature branch against the `dev` branch instead of the `master` branch, you probably would not have this problem.

You can resolve by changing all of this:

```
<<<<<<< master
	$(TINYGO) build -size short -o test.hex -target=pygamer             examples/blinky1
	@$(MD5SUM) test.hex
=======
>>>>>>> dev
```

to this:

```
	$(TINYGO) build -size short -o test.hex -target=pygamer             examples/blinky1
	@$(MD5SUM) test.hex
```

You could use the web editor or however you normally edit code.

Hope that helps!

APDevice committed:
added pygamer to smoketest

APDevice committed:
added pygamer to board

APDevice committed:
Adding Board: PyGamer

APDevice committed:
resolve conflict between master and dev

APDevice commented:
Well the confusing thing is that that’s not in the file. I even double checked. 

Ok, so I merged my dev with my master. It’s sloppy I realize, but it seems to resolve the issue.

APDevice committed:
Merge pull request #1 from APDevice/dev

merging dev and master

APDevice committed:
corrected missing extension

deadprogram commented:
Thank you very  much for working on this @APDevice I will now squash/merge into `dev`. Thanks again!
